root/components/common-macros.tt

```diff
@@ -5,17 +5,11 @@
 [%~ MACRO add_commas(n) GET n.chunk(-3).join(',') -%]
 [%~ MACRO replace(text, search, replace) BLOCK; text | replace(search, replace); END -%]
 [%~ MACRO url_escape(url) BLOCK; url | url; END -%]
-[%~ MACRO html_escape(url) BLOCK; url | html; END -%]
+[%~ MACRO html_escape(text) BLOCK; text | html; END ~%]
 [%~ MACRO html_unescape(text) BLOCK; text.replace('&quot;', '"').replace('&lt;', '<').replace('&gt;', '>').replace('&#39;', "'").replace('&amp;', '&'); END -%]
 [%~ MACRO bracketed(text) BLOCK; text != '' ? l(' ({text})', { text => text }) : ''; END -%]
 [%~ MACRO js_escape(text) BLOCK; text | js; END ~%]
 [%~ MACRO closing_tag_escape(text) BLOCK; text.replace('</', '<\\/'); END ~%]
-[%~ MACRO display_url(url) BLOCK; PROCESS _unescaped_display_url | html; END -%]
-[%~ BLOCK _unescaped_display_url;
-    IF url.pretty_name; url.pretty_name;
-    ELSE; url | uri_decode;
-    END;
```

Comment:
Essentially, this distinguished between URL entities (having a `pretty_name` method) and other URI objects. For the former, I just replaced the macro call by `x.pretty_name`; for the latter (of which there only was one case), we can let the URI object do the decoding.